src/H5Dchunk.c

```diff
@@ -8222,20 +8234,16 @@ H5D__chunk_iter(H5D_t *dset, H5D_chunk_iter_op_t op, void *op_data)
 } /* end H5D__chunk_iter() */
 
 /*-------------------------------------------------------------------------
- * Function:    H5D__chunk_get_offset_copy
+ * Function:    H5D__chunk_verify_offset
```

Comment:
Turns out nothing in the direct chunk code modifes the offset array so there's no need to copy it and we can just pass it directly from the user qualified as const. Added this function to perform verification.